src/foam/nanos/geocode/GoogleMapsGeocodingDAO.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.geocode;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.lib.json.JSONParser;
+import foam.nanos.auth.Address;
+import foam.nanos.auth.User;
+import foam.util.SafetyUtil;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class GoogleMapsGeocodingDAO
```

Comment:
If this only works for Users I would put that in the name. Or better, provide the PropertyInfo for the Address field so it can be used for any Model with an Address.

Reply:
Modified it to accept PropertyInfo and modified the put_ method to use that instead of checking if it's a user or not.